File: Problems/897. Increasing Order Search Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<TreeNode> inorderNodes;
    void inorderTraversal(TreeNode* root){
        if(root == NULL)
            return;
        inorderTraversal(root->left);
        inorderNodes.push_back(root->val);
        inorderTraversal(root->right);
    }
    
    TreeNode* increasingBST(TreeNode* root) {
        inorderTraversal(root);
        TreeNode* curr = new TreeNode(inorderNodes[0]);
        TreeNode* first = curr;
        
        for(int i=1; i<inorderNodes.size(); i++){
            curr->right = new TreeNode(inorderNodes[i]);
            curr = curr->right;
        }
        
        return first;
    }
};

File: Problems (Python)/897. Increasing Order Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:    
    def increasingBST(self, root: TreeNode) -> TreeNode:
        inNodes = []
        def inorderTraversal(root: TreeNode):
            if root:
                inorderTraversal(root.left)
                inNodes.append(root.val)
                inorderTraversal(root.right)
            
        inorderTraversal(root)
        curr = first = TreeNode(inNodes[0])
        for i in range(1, len(inNodes)):
            curr.right = TreeNode(inNodes[i])
            curr = curr.right
        
        return first
